src/libraries/Microsoft.PowerFx.Core/Syntax/UserDefinitions.cs

```diff
@@ -155,10 +177,11 @@ private IEnumerable<UserDefinedFunction> CreateUserDefinedFunctions(IEnumerable<
             return userDefinedFunctions;
         }
 
-        private bool CheckParameters(ISet<UDFArg> args, List<TexlError> errors)
+        private bool CheckParameters(ISet<UDFArg> args, List<TexlError> errors, INameResolver nameResolver, out List<DType> parameterTypes)
         {
             var isParamCheckSuccessful = true;
             var argsAlreadySeen = new HashSet<string>();
+            parameterTypes = new List<DType>(); 
```

Comment:
>parameterTypes = new List<DType>();  [](http://example.com/codeflow?start=12&length=36)

You can add 

if(args.Count == 0)
   return false;

No need to allocate list in that case.